frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -130,24 +130,24 @@ end
 
 -- Generate all book status file dialog buttons in a row
 function filemanagerutil.genStatusButtonsRow(doc_settings_or_file, caller_callback)
-    local summary, status
+    local file, summary, status
     if type(doc_settings_or_file) == "table" then -- currently opened file
+        file = doc_settings_or_file:readSetting("doc_path")
         summary = doc_settings_or_file:readSetting("summary")
         status = summary.status
     else
-        status = filemanagerutil.getStatus(doc_settings_or_file)
+        file = doc_settings_or_file
+        summary = {}
+        status = filemanagerutil.getStatus(file)
     end
     local function genStatusButton(to_status)
         return {
             text = filemanagerutil.statusToString(to_status) .. (status == to_status and "  ✓" or ""),
-            id = to_status, -- used by covermenu
             enabled = status ~= to_status,
             callback = function()
-                if summary then -- currently opened file
-                    summary.status = to_status
-                else
-                    filemanagerutil.setStatus(doc_settings_or_file, to_status)
-                end
+                summary.status = to_status
+                filemanagerutil.setStatus(file, to_status)
+                UIManager:broadcastEvent(Event:new("DocSettingsChanged", file, { summary = summary }))
```

Comment:
This name now feels too generic :) It feels we'd need to add it in thousand places.
But ok, I guess you'll just need to emit it when `some` specific docsettings item changed, and it might need a comment above saying for what other part of the code this one is needed (`-- Needed by CoverBrowser`, which goes against mentionning CoverBrowser in core, but well :)).
May be name it `DocSettingsItemsUpdated` to make it clearer that a table with only the key(s)/value(s) of things that changed is send with it?

(May be just me, but `Updated` sounds better (or is easier to pronounce) than `Changed`.)

Reply:
The name is alike
https://github.com/koreader/koreader/blob/a84ab265b8626791244700adb0a206755d365bd6/frontend/apps/filemanager/filemanagerbookinfo.lua#L176